Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -215,19 +241,17 @@ def Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
 def Test(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
     ShowMof('TEST', ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
     retval = 0
-    if not os.path.exists(AGENT_SCRIPT_PATH):
-        LG().Log(LogType.Error, 'npmd set cap script does not exist, exiting test')
+    if not os.path.exists(PATH_SETCAP_SCRIPT):
+        LOG_ACTION.log(LogType.Error, 'npmd set cap script does not exist, exiting test')
         return [retval]
     if ConfigType != 'UpdatedAgentConfig':
-        LOG_ACTION.log(LogType.Error, 'Config type did not match, exiting test')
+        LOG_ACTION.log(LogType.Error, 'Config type did not match, exiting test for workspace ' + ConfigID)
         return [retval]
-    if Ensure == 'Absent':
-        if os.path.exists(AGENT_RESOURCE_VERSION_PATH):
-            LG().Log(LogType.Info, 'Ensure is absent, resource is present on the agent, set will purge')
-            retval = -1
+    if TestMultiHoming(ConfigID, Ensure) != 0:
```

Comment:
The way you check for non-zero return codes is little inconsistent; it would be clearer (to me at least) if you changed them all to "!= 0" checks.